### PR TITLE
Adding fields to job_metadata for workflows and approval nodes

### PR DESCRIPTION
**Pull request by john-westcott-iv**
##### SUMMARY

Adding extra vars for workflow approval notifications.
Adding the following for workflow templates:
  * inventory
  * limit
  * extra_vars

Added empty extra_vars to the content_stub.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 21.0.1.dev123+g6587e95783
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: awx/main/models/workflow.py [hunk 1]
```diff
@@ -885,3 +892,12 @@ def workflow_job_template(self):
     @property
     def workflow_job(self):
         return self.unified_job_node.workflow_job
+
+    def notification_data(self):
+        result = super(WorkflowApproval, self).notification_data()
+        result.update(
+            dict(
+                extra_vars=self.workflow_job.extra_vars,
```
Comment (AlanCoding) [inline awx/main/models/workflow.py]: wait a minute, you used `self.display_extra_vars()` for workflow notification data, but then `self.workflow_job.extra_vars` for the approval node notification data. Shouldn't this be `self.workflow_job.display_extra_data()`?

Comment (john-westcott-iv) [reply awx/main/models/workflow.py]: Good catch, we want display_ here since stuff could bleed through the workflow.

**Review by AlanCoding — APPROVED** [review]

**Comment by AlanCoding** [comment]
This still might need some :brain: power for docs, but it seems to solve the ask.

We should first figure out what content we already have for notification templating rules for workflows & approval nodes. If that's scantly documented to begin with, communicating this could be a challenge.

File: awx/main/models/workflow.py [hunk 2]
```diff
@@ -659,6 +659,13 @@ def notification_data(self):
                 node_job_description = 'job #{0}, "{1}", which finished with status {2}.'.format(node.job.id, node.job.name, node.job.status)
             str_arr.append("- node #{0} spawns {1}".format(node.id, node_job_description))
         result['body'] = '\n'.join(str_arr)
+        result.update(
```
Comment (jbradberry) [inline awx/main/models/workflow.py:662]: `dict.update()` does take kwargs, so constructing another inner `dict` isn't necessary

Comment (jbradberry) [reply awx/main/models/workflow.py:662]: As an aside, I want to advocate for our team to get away from the style of using manual `dict()` calls in favor of `dict` literals.  It's less flexible.

**Review by jbradberry — APPROVED** [review]

**Comment by lucab85** [comment]
can we merge?

**Review by lucab85 — APPROVED** [review]

**Comment by jbradberry** [comment]
I believe we can merge, yes.

**Pull request merged**